pkg/wharfapi/test_result.go

```diff
@@ -80,3 +83,30 @@ func (c *Client) GetBuildAllTestResultListSummary(buildID uint) (response.TestRe
 	err := c.getUnmarshal(path, nil, &listSummary)
 	return listSummary, err
 }
+
+// CreateBuildTestResult uploads a test result file (eg: "tests.trx") by
+// invoking the HTTP request:
+//  POST /api/build/{buildId}/test-result
+//
+// Added in wharf-api v5.0.0.
+func (c *Client) CreateBuildTestResult(buildID uint, fileName string, testResult io.Reader) ([]response.ArtifactMetadata, error) {
+	if err := c.validateEndpointVersion(5, 0, 0); err != nil {
+		return nil, err
+	}
+	path := fmt.Sprintf("/api/build/%d/test-result/", buildID)
```

Comment:
nit: No other path has trailing slash.
```suggestion
	path := fmt.Sprintf("/api/build/%d/test-result", buildID)
```

Reply:
This is actually a requirement, because in wharf-api this endpoint in particular is registered on `"/"` and not `""`.

Gin adds a redirect, but for some reason wharf-api-client-go doesn't follow redirects. I did not have the energy to look into it, so I just specified the non-redirecting link instead.

I've now created a separate issue on this: #41

Reply:
Ah I see 👍🏻